terraform_modules/states_framework/backend.tf

```diff
@@ -0,0 +1,68 @@
+resource aws_s3_bucket tfstates {
+  bucket              = "nr-coreint-canaries-tfstates"
+  object_lock_enabled = true
+  force_destroy     = true # Allow to destroy all the versions automatically created.
```

Comment:
I would comment this line so the s3 bucket cannot be destroyed by mistake.

This is by far the most critical resource created by this PR.

Reply:
Good point 👍